src/GameWindow.py

```diff
@@ -0,0 +1,106 @@
+    def __init__(self, parent=None):
+        super(GameWindow, self).__init__(parent)
+        self.ui = Ui_GameWindow()
+        self.ui.setupUi(self)
+        self.setStyleSheet('''#centralWidget{
+        border-image: url(ship.jpg) 0 0 0 0 stretch stretch;
+        }
+        QLabel{
+        color: #FFA500;
+        font-weight: bold;
+        }
+        ''')
+
+        self.ui.actionPreparePlayer_1.triggered.connect(self.prepare_player_1)
+        self.ui.actionPreparePlayer_2.triggered.connect(self.prepare_player_2)
+        self.ui.actionPlayer_1.triggered.connect(self.player_1)
+        self.ui.actionPlayer_2.triggered.connect(self.player_2)
+
+    def mousePressEvent(self, event):
+        # вдруг пригодится
+        logging.debug("Game window mouse press")
+        super(GameWindow, self).mousePressEvent(event)
+
+    def set_widgets_on_off(self, is_first: bool = True):
+        self.ui.wigdet_player_1.setEnabled(is_first)
+        self.ui.wigdet_player_2.setEnabled(not is_first)
+
+    def hide_ships(self):
+        self.ui.ships_list_1.hide()
+        self.ui.ships_list_2.hide()
+
+    def prepare_player_1(self):
+        logger = logging.getLogger(__name__)
+        logger.debug('prepare_player_1')
+
+        self.ui.label_player_1.setText('[Prepare player 1]')
+        self.ui.label_player_2.setText('Prepare player 2')
+
+        self.set_widgets_on_off(is_first=True)
+
+    def prepare_player_2(self):
+        logger = logging.getLogger(__name__)
+        logger.debug('prepare_player_2')
+        self.ui.label_player_1.setText('Prepare player 1')
+        self.ui.label_player_2.setText('[Prepare player 2]')
+
+        self.set_widgets_on_off(is_first=False)
+
+    def player_1(self):
+        logger = logging.getLogger(__name__)
+        logger.debug('player_1')
+
+        self.ui.label_player_1.setText('[Player 1]')
+        self.ui.label_player_2.setText('Player 2')
+
+        self.hide_ships()
+
+        self.set_widgets_on_off(is_first=True)
+        self.ui.wigdet_player_1.setGraphicsEffect(None)
+        self.ui.wigdet_player_2.setGraphicsEffect(QGraphicsBlurEffect())
+
+    def player_2(self):
+        logger = logging.getLogger(__name__)
+        logger.debug('player_2')
+
+        self.ui.label_player_1.setText('Player 1')
+        self.ui.label_player_2.setText('[Player 2]')
+
+        self.hide_ships()
+
+        self.set_widgets_on_off(is_first=False)
+        self.ui.wigdet_player_1.setGraphicsEffect(QGraphicsBlurEffect())
+        self.ui.wigdet_player_2.setGraphicsEffect(None)
+
+    def state_machine(self, type_):
+        if type_ == States.PREPARE_PLAYER_1:
+            self.prepare_player_1()
+        elif type_ == States.PREPARE_PLAYER_2:
+            self.prepare_player_2()
+        elif type_ == States.MOVE_PLAYER_1:
+            self.player_1()
+        elif type_ == States.MOVE_PLAYER_2:
+            self.player_2()
+        else:
+            raise Exception(f'Unknown type: {type_}')
```

Comment:
Теперь по вызову Next() твой класс способен сам понимать, что и какому игроку надо обеспечить. Только нужно заранее сохранить то, какой игрок сейчас что-то делает.

```suggestion
    def Next(self):
        if self.currentState == States.UNDEFINED:
            self.currentPlayer = random.randint(1, 2)
            self.currentState = States.FIRST_PREPARE
            self.PreparePlayer(self.currentPlayer)

        elif self.currentState == States.FIRST_PREPARE:
            self.currentPlayer = 2 if self.currentPlayer == 1 else 1
            self.currentState = States.SECOND_PREPARE
            self.PreparePlayer(self.currentPlayer)

        elif self.currentState == States.SECOND_PREPARE:
            self.currentPlayer = random.randint(1, 2)
            self.currentState = States.GAME
            self.MovePlayer(self.currentPlayer)

        elif self.currentState == States.GAME:
            self.currentPlayer = 2 if self.currentPlayer == 1 else 1
            self.MovePlayer(self.currentPlayer)
```

Reply:
Ну я первый раз с ней работаю

_Уф, нейминг методов, зубы скрепят..._

Reply:
Такс, а если я попал в корабль? Я не хочу ход отдавать(

Reply:
А, внатуре, ну тогда аргумент методу задай, типа keep_player

Reply:
Вот вот(

Надо подумать

Reply:
`def next(self, keep_player: bool = False)` Типа такого?

Reply:
Да